src/operators/arrange/arrangement.rs

```diff
@@ -690,30 +667,37 @@ where
     }
 }
 
-impl<G: Scope, K: ExchangeData+Hashable, R: ExchangeData+Semigroup> Arrange<G, K, (), R> for Collection<G, K, R>
-where
-    G::Timestamp: Lattice+Ord,
-{
-    fn arrange_core<P, Tr>(&self, pact: P, name: &str) -> Arranged<G, TraceAgent<Tr>>
-    where
-        P: ParallelizationContract<G::Timestamp, ((K,()),G::Timestamp,R)>,
-        Tr: Trace+TraceReader<Time=G::Timestamp>+'static,
-        Tr::Batch: Batch,
-        Tr::Batcher: Batcher<Item = ((K,()),G::Timestamp,R), Time = G::Timestamp>,
-        Tr::Builder: Builder<Item = ((K,()),G::Timestamp,R), Time = G::Timestamp, Output = Tr::Batch>,
-    {
-        self.map(|k| (k, ()))
-            .arrange_core(pact, name)
-    }
-}
```

Comment:
This is too bad, but roughly there is a constraint satisfaction issue that Rust can't seem to determine that `(_, ())` can't unify with `_`, which it was previously able to determine for `_ = K`, but cannot when `_ = K::Owned`. It's possible that I could fix something here, but one can work around this by changing the call to `arrange` on non-pair data into `arrange_by_self`, which is the more explicit way to arrange that way.

Reply:
This got fixed!